furiousatoms/molecular.py

```diff
@@ -1,110 +1,148 @@
 
 from furiousatoms.structure import bbox
-import MDAnalysis
-from fury import window, actor, ui, molecular as mol
-import fury.primitive as fp
-
-table = mol.PTable()
-# TODO: I have to replace this class with the ViewerMemoryManager class
-# that does not depend on universe (md analysis).
-class UniverseManager:
-
-    def __init__(self, universe, no_bonds=0):
+from fury import actor, molecular as mol
+
+periodic_table = PTable()
+
+SPHERE_SIZE = 0.35
+POS_DIM = 3
+BOND_DIM = 2
+
+class MolecularStructure:        
+    def __init__(self, box_size: list, pos: np.array, bonds: np.array, atom_types: np.array):
+        '''
+        `box_size`: a 1x3 list with X, Y, and Z numerical dimensions. 
+        `pos`: a Nx3 array with XYZ float positions of each atom.
+        `bonds`: a Nx2 array where each row is a pair of integer atom IDs. 
+            Each ID corresponds to a row in `pos`.
+        `atom_types`: a Nx1 array of strings representing element symbols. 
+            The length must match with `pos`.
+        '''
+        self.box_size = box_size
+        self.pos = pos
+        self.bonds = bonds
+        self.atom_types = atom_types
+
+    def create_empty():
+        box_size = [0, 0, 0]
+        pos = np.empty(shape=(0, 3))
+        bonds = np.empty(shape=(0, 2), dtype=int)
+        atom_types = np.empty(shape=(0), dtype=str)
+        return MolecularStructure(box_size, pos, bonds, atom_types)
+
+    def merge(s1, s2, offset_bonds=False):
+        '''
+        Merge a MolecularStructure with another, combining the positions, bonds, and atom_types.
+        Only s1's box_size will be kept. Neither s1 nor s1 will be modified. 
+        `offset_bonds`: If set to True, bond IDs will be corrected so that the two structures
+            will not connect.
+        '''
+        pos_merged = np.zeros(shape=(len(s1.pos) + len(s2.pos), 3))
+        i = 0
+        for position in s1.pos:
+            for j in range(3):
+                pos_merged[i][j] = position[j]
+            i += 1
+        for position in s2.pos:
+            for j in range(3):
+                pos_merged[i][j] = position[j]
+            i += 1
+
+        offset = s1.pos.shape[0] if offset_bonds else 0
+        bonds_merged = np.zeros(shape=(len(s1.bonds) + len(s2.bonds), 2), dtype=int)
+        i = 0
+        for bond in s1.bonds:
+            for j in range(2):
+                bonds_merged[i][j] = bond[j]
+            i += 1
+        for bond in s2.bonds:
+            for j in range(2):
+                bonds_merged[i][j] = bond[j] + offset
+            i += 1
+
+        types_merged = np.zeros(shape=(len(s1.atom_types) + len(s2.atom_types)), dtype=str)
+        i = 0
+        for typ in s1.atom_types:
+            types_merged[i] = typ
+            i += 1
+        for typ in s2.atom_types:
+            types_merged[i] = typ
+            i += 1
+
+        return MolecularStructure(s2.box_size, pos_merged, bonds_merged, types_merged)
+
+    def center(self):
```

Comment:
`center()` is in-place whereas `merge` is not. Example:
```
structure = structure.merge(other_structure) #This creates a new structure instead of modifying either one
structure.center() #This actually modifies the structure
```
I feel this is inconsistent, but it is better for performance. Is this okay?